src/simulation/m_monopole.fpp

```diff
@@ -165,119 +180,107 @@ contains
-                        the_time = t_step*dt
-                        if ((the_time >= delay(q)) .or. f_is_default(delay(q))) then
-                            !$acc loop seq
-                            do ii = 1, num_fluids
-                                myalpha_rho(ii) = q_cons_vf(ii)%sf(j, k, l)
-                                myalpha(ii) = q_cons_vf(advxb + ii - 1)%sf(j, k, l)
-                            end do
+                        the_time = t_step*dt ! TODO move outside of the loop?
+                        if (the_time < delay(q) .and. pulse(q) == 1) cycle ! need to explicitly set delay to 0 if default; now it hinges on the fact default is a negative value
 
-                            myRho = 0d0
-                            n_tait = 0d0
-                            B_tait = 0d0
-
-                            if (bubbles) then
-                                if (mpp_lim .and. (num_fluids > 2)) then
-                                    !$acc loop seq
-                                    do ii = 1, num_fluids
-                                        myRho = myRho + myalpha_rho(ii)
-                                        n_tait = n_tait + myalpha(ii)*gammas(ii)
-                                        B_tait = B_tait + myalpha(ii)*pi_infs(ii)
-                                    end do
-                                else if (num_fluids > 2) then
-                                    !$acc loop seq
-                                    do ii = 1, num_fluids - 1
-                                        myRho = myRho + myalpha_rho(ii)
-                                        n_tait = n_tait + myalpha(ii)*gammas(ii)
-                                        B_tait = B_tait + myalpha(ii)*pi_infs(ii)
-                                    end do
-                                else
-                                    myRho = myalpha_rho(1)
-                                    n_tait = gammas(1)
-                                    B_tait = pi_infs(1)
-                                end if
-                            else
+                        !$acc loop seq
+                        do ii = 1, num_fluids
+                            myalpha_rho(ii) = q_cons_vf(ii)%sf(j, k, l)
+                            myalpha(ii) = q_cons_vf(advxb + ii - 1)%sf(j, k, l)
+                        end do
+
+                        myRho = 0d0
+                        n_tait = 0d0
+                        B_tait = 0d0
+
+                        if (bubbles) then
+                            if (mpp_lim .and. (num_fluids > 2)) then
                                 !$acc loop seq
                                 do ii = 1, num_fluids
                                     myRho = myRho + myalpha_rho(ii)
                                     n_tait = n_tait + myalpha(ii)*gammas(ii)
                                     B_tait = B_tait + myalpha(ii)*pi_infs(ii)
                                 end do
+                            else if (num_fluids > 2) then
```

Comment:
So long as you are changing things, it seems like this part of the code that computes the speed of sound `sound` should either be computed by `s_compute_speed_of_sound` or, if that isn't the right routine for some reason, an abstracted-out subroutine

Reply:
You are right. I kept it unchanged, but should have switched to using `s_compute_speed_of_sound`. I've checked and they produce the same result. It'll need a few more lines before calling `s_compute_speed_of_sound`, and I can abstract out all these into a subroutine.

Reply:
this sounds like a plan!